src/components/posts/ModerationProvider.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { ReactNode, useEffect, useMemo, useRef, useState } from 'react'
+import { isDevMode } from 'src/config/env'
+import { parseGrillMessage } from 'src/utils/iframe'
+import { getCurrentUrlOrigin } from 'src/utils/url'
+
+type State = {
+  openModal: (postId: string) => void
+}
+
+const Context = React.createContext<State>({ openModal: () => undefined })
+
+export default function ModerationProvider({ children }: { children: ReactNode }) {
+  const [isOpenModal, setIsOpenModal] = useState(false)
+  const iframeRef = useRef<HTMLIFrameElement | null>(null)
+
+  useEffect(() => {
+    const listener = (event: MessageEvent<any>) => {
+      const message = parseGrillMessage(event.data + '')
+      if (!message) return
+
+      const { name, value } = message
+      if (name === 'moderation' && value === 'close') {
+        setIsOpenModal(false)
+      }
+    }
+    window.addEventListener('message', listener)
+    return () => window.removeEventListener('message', listener)
+  }, [])
+
+  const value = useMemo(() => {
+    return {
+      openModal: (postId: string) => {
+        iframeRef.current?.contentWindow?.postMessage(
+          {
+            type: 'grill:moderate',
+            payload: postId,
+          },
+          '*',
+        )
+        setIsOpenModal(true)
+      },
+    }
+  }, [])
+
+  return (
+    <Context.Provider value={value}>
+      {children}
+      {!isDevMode && (
+        <iframe
```

Comment:
I hope we are rendering one iframe in one time?

Because if we have 

- moderation
- my profile
- chat creation 

modals, and didnt clear it, we keep 3 grill iframes in one time

Reply:
right now we have 2 modals, for profile and moderation (only if user is super admin)
chat creation is for later, but yeah make sense to refactor it from now

Reply:
Yes, lets create task for it in "SHOULD"

Reply:
oh, so should I refactor it now? or at later after chat creation?